### PR TITLE
Add IPv6 tests for all transports

### PR DESCRIPTION
**Pull request by KushnerykPavel**
Resolve https://github.com/iggy-rs/iggy/issues/484

### CODE REVIEW

File: integration/tests/bench/http.rs [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
 }
+
+#[test]
+#[parallel]
+fn http_ipv6_bench() {
+    let mut test_server = TestServer::new(None, true, None, IpAddrKind::V6);
+    test_server.start();
+    let server_addr = test_server.get_http_api_addr().unwrap();
+    run_bench_and_wait_for_finish(&server_addr, Transport::Http);
+}
```
Comment (BartoszCiesla) [inline integration/tests/bench/http.rs]: Missing empty line at the end of file

File: integration/tests/bench/quic.rs [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
+
+
+#[test]
+#[parallel]
+fn quic_ipv6_bench() {
+    let mut test_server = TestServer::new(None, true, None, IpAddrKind::V6);
+    test_server.start();
+    let server_addr = test_server.get_quic_udp_addr().unwrap();
+    run_bench_and_wait_for_finish(&server_addr, Transport::Quic);
+}
```
Comment (BartoszCiesla) [inline integration/tests/bench/quic.rs]: Missing empty line at the end of file

File: integration/tests/bench/tcp.rs [hunk 3]
```diff
@@ -1,12 +1,22 @@
...
+
+
+#[test]
+#[parallel]
+fn tcp_ipv6_bench() {
+    let mut test_server = TestServer::new(None, true, None, IpAddrKind::V6);
+    test_server.start();
+    let server_addr = test_server.get_raw_tcp_addr().unwrap();
+    run_bench_and_wait_for_finish(&server_addr, Transport::Tcp);
+}
```
Comment (BartoszCiesla) [inline integration/tests/bench/tcp.rs]: Missing empty line at the end of file

**Review by BartoszCiesla — COMMENTED** [review]
Please check also verification failures

File: integration/src/test_server.rs [hunk 4]
```diff
@@ -252,6 +262,15 @@ impl TestServer {
         ]
     }
 
+    fn get_server_ipv6_addrs_with_random_port() -> Vec<ServerProtocolAddr> {
+        let addr = SocketAddr::new(Ipv6Addr::new(0, 0, 0, 0, 0, 0, 0, 1).into(), 0);
```
Comment (hubcio) [inline integration/src/test_server.rs]: make it `Ipv6Addr::LOCALHOST`

**Review by hubcio — DISMISSED** [review]

**Comment by KushnerykPavel** [comment]
@hubcio can you please, help me figure out with linux test failure? I do not see any errors in logs.

**Comment by KushnerykPavel** [comment]
I ran next command 
`cargo r --bin iggy-bench send-and-poll --messages-per-batch 100 --message-batches 100 --message-size 1000 quic --server-address ::1:3353`
and get next error 
`
thread 'main' panicked at bench/src/args/common.rs:63:74:
called `Result::unwrap()` on an `Err` value: AddrParseError(Socket)
`

Can we probably enrich bench output in future?

**Comment by KushnerykPavel** [comment]
I have seen potential problem here https://github.com/iggy-rs/iggy/blob/master/integration/src/quic_client.rs#L18 

We use default config with client ipv4 address and server ipv6 address.

**Comment by hubcio** [comment]
@KushnerykPavel is everything OK now?

have you already joined our discord?

**Comment by KushnerykPavel** [comment]
@hubcio Yes, sir! Could you also check changes in iggy/src/quic/client.rs . Is it looks good for you ?

File: iggy/src/quic/client.rs [hunk 5]
```diff
@@ -157,8 +157,16 @@ impl QuicClient {
...
         let server_address = config.server_address.parse::<SocketAddr>()?;
+        let mut client_parsed_address = config.client_address.to_string();
+
+        if server_address.is_ipv6()
+            && config.client_address == QuicClientConfig::default().client_address
+        {
+            client_parsed_address = "[::1]:0".to_string();
+        }
+
+        let client_address = client_parsed_address.parse::<SocketAddr>()?;
```
Comment (hubcio) [inline iggy/src/quic/client.rs]: more idiomatic approach:

```rust
    let client_address = if server_address.is_ipv6() && config.client_address == default_client_address {
        "[::1]:0"
    } else {
        &config.client_address
    }
    .parse::<SocketAddr>()
    .map_err(|_| Error::InvalidClientAddress)?;
    ```

**Review by hubcio — DISMISSED** [review]

**Comment by spetz** [comment]
Looks good, just one more unused import to remove :)

**Review by spetz — APPROVED** [review]

**Pull request merged**